src/sinks/aws_s3.rs

```diff
@@ -298,6 +290,10 @@ impl Service<Request> for S3Sink {
         }
         let tagging = tagging.finish();
 
+        emit!(S3EventsSent {
+            byte_size: request.body.len(),
+        });
```

Comment:
🤔 I think we typically don't emit these metrics unless the sending is successful.

Reply:
In practice, we do... which I don't agree with.  I'm happy to shift it to a position where it is more representative of where/when it leaves Vector, but was initially going for consistency.

(As an example, the Elasticsearch sink emits this metric simply when the log event is encoded, and it happens similarly in the generic HTTP sink, etc.)

Reply:
Indeed, you are right. I think this is OK for now then to maintain consistency.

Reply:
Yeah, I ran into similar problems trying to instrument some other sinks and being unsure where to actually emit these